src/analysis/modifications/moment.jl

```diff
@@ -0,0 +1,110 @@
+"""
+add_moment_constraints(
+    kcats::Dict{String,Float64},
+    molar_weights::Dict{String,Float64},
+    protein_mass_fraction::Float64;
+    protected_constraint_ids = ["ATPM"],
+)
+
+Add enzyme capacity constraints to problem using the MOMENT algorithm. Requires `kcats` and
+`molar_weights` for each reaction and protein. Proteins are encoded by their associated gene.
+Adds a variable vector `y` to the problem corresponding to the protein concentration. Adds a
+variable vector `t` that is internal to the problem - ignore. Set the `protein_mass_fraction`
+capacity constraint to the fraction of the cell mass that is composed of proteins.
+Optionally `protected_constraint_ids` can be used to ensure that some reaction fluxes are not
+unconstrained; all reaction `ids` not protected are set as unconstrained in the direction they
+are set to in the original model. Exchange reactions are protected by default.
+
+See Adadi, Roi, et al. "Prediction of microbial growth rate versus biomass yield by a
+metabolic network with kinetic parameters." PLoS computational biology (2012) for more
+details.
+
+# Example
+```
+sol = flux_balance_analysis_dict(
+    model,
+    Tulip.Optimizer;
+    modifications = [
+        add_moment_constraints(
+            kcats,
+            molar_weights,
+            protein_mass_fraction;
+            protected_constraint_ids = protected_constraint_ids,
+        ),
+        change_constraint("EX_glc__D_e", lb = -1000),
+    ],
+)
+```
+"""
+add_moment_constraints(
+    kcats::Dict{String,Float64},
+    molar_weights::Dict{String,Float64},
+    protein_mass_fraction::Float64;
+    protected_constraint_ids = ["ATPM"],
```

Comment:
this might be a bit surprising, I'd vote for empty defaults instead and documenting that whoever forgets about ATP stays without energy for a week. :D